distributed/tests/test_client.py

```diff
@@ -4038,15 +4039,14 @@ async def test_serialize_future(s, a, b):
         result = await future
 
         for ci in (c1, c2):
-            for ctxman in lambda ci: ci.as_current(), lambda ci: temp_default_client(
-                ci
-            ):
-                with ctxman(ci):
-                    future2 = pickle.loads(pickle.dumps(future))
-                    assert future2.client is ci
-                    assert stringify(future2.key) in ci.futures
-                    result2 = await future2
-                    assert result == result2
+            with ci.as_current():
+                future2 = pickle.loads(pickle.dumps(future))
+                assert future2.client is ci
+                assert stringify(future2.key) in ci.futures
+                result2 = await future2
+                assert result == result2
+            with temp_default_client(ci), pytest.raises(NoCurrentClient):
+                future2 = pickle.loads(pickle.dumps(future))
```

Comment:
There is a subtle difference between default clients and current clients. Current client is stricter and better in almost all circumstances. temp_default is also only used in testing. 
I think this change makes everything much more predictable and _should_ not have an effect on actual UX